airflow/models/taskinstance.py

```diff
@@ -374,14 +375,12 @@ def log_filepath(self):
 
     @property
     def log_url(self):
-        iso = quote(self.execution_date.isoformat())
-        base_url = configuration.conf.get('webserver', 'BASE_URL')
-        return base_url + (
-            "/log?"
-            "execution_date={iso}"
-            "&task_id={task_id}"
-            "&dag_id={dag_id}"
-        ).format(iso=iso, task_id=self.task_id, dag_id=self.dag_id)
+        url = url_for(
```

Comment:
I think using url_for is going to break some uses of this - namely in failure emails, where there is no active flask request, so url_for won't work.

Reply:
Ah I understand. Let me check if it also works in the email alert.

Reply:
Oh also - this url is going to be wrong for RBAC too :( (`Airflow.log` is what we need for RBAC)

Reply:
I'm not sure I like it, but:

```python
if not flask._request_ctx_stack:
    with request_context(... MESSY_ARGS ... ):
       return url_for(...)
else:
    return url_for(...)
```

Also actually making the args to work for `request_context()` may be hard (not to mention messy in a model.)

Reply:
`RuntimeError: Attempted to generate a URL without the application context being pushed. This has to be executed when application context is available.` Thank you. :)

Reply:
> Oh also - this url is going to be wrong for RBAC too :( (Airflow.log is what we need for RBAC)

Yes, I noticed that, too.
So we probably should not try to access the request context in the task instance. But for the `log_url_formatter` it should work and since we have a rbac view and a non-rbac view we can use `Airflow.log` and `airflow.log`

Reply:
Yeah - that would fix the column in task instances view at least

Reply:
> Also actually making the args to work for request_context() may be hard (not to mention messy in a model.)

Agree. This would be messy. I don't like that.
So we can only let it be like it is and differentiate depending on the value in the airflow config `rbac=True/False` to build the url either with `airflow/admin` or not?!

Reply:
So like that:

```python
    @property
    def log_url(self):
        iso = quote(self.execution_date.isoformat())
        base_url = configuration.conf.get('webserver', 'BASE_URL')

        relative_url = "/log?execution_date={iso}&task_id={task_id}&dag_id={dag_id}".format(
            iso=iso, task_id=self.task_id, dag_id=self.dag_id)

        if configuration.conf.get('webserver', 'rbac'):
            return base_url + relative_url
        return base_url + '/admin/airflow' + relative_url
```